connect-inject/container_init.go

```diff
@@ -127,12 +139,19 @@ func (h *Handler) containerInit(pod *corev1.Pod) (corev1.Container, error) {
 			}
 
 			if port > 0 {
-				data.Upstreams = append(data.Upstreams, initContainerCommandUpstreamData{
+				upstream := initContainerCommandUpstreamData{
 					Name:       service_name,
 					LocalPort:  port,
 					Datacenter: datacenter,
 					Query:      prepared_query,
-				})
+				}
+
+				// Add namespace to upstream
+				if namespace != "" {
+					upstream.ConsulUpstreamNamespace = namespace
+				}
```

Comment:
Do we need this `if`? If `ConsulUpstreamNamespace` is an empty string, then the template ` {{- if .ConsulUpstreamNamespace }}` below will evaluate to `false` I believe.